File: website/src/app/shared/learn/models/learn-task.ts
import {LearnSubTask} from './learn-sub-task';
import {SignDetailsModel} from '../../signs/models/sign-details.model';

export class LearnTask {
  currentLearnTaskIndex = 0;
  learnTasks: LearnSubTask[];

  deserialize(input: any): this {
    Object.assign(this, input);
    for (let i = 0; i < this.learnTasks.length; i++) {
      this.learnTasks[i] = new LearnSubTask().deserialize(this.learnTasks[i]);
    }
    return this;
  }

  getCurrentLearnTask() {
    return this.learnTasks[this.currentLearnTaskIndex];
  }

  getLearnedSigns(): SignDetailsModel[] {
    const learnedSigns = [];
    for (const learnSubTask of this.learnTasks) {
      learnedSigns.push(learnSubTask.question);
    }
    return learnedSigns;
  }
}

File: api/src/main/java/com/newmedia/deafapi/api/models/LearnTask.java
package com.newmedia.deafapi.api.models;

public class LearnTask {

    private LearnSubTask[] learnTasks;

    public LearnTask(LearnSubTask[] learnTasks) {
        this.learnTasks = learnTasks;
    }

    public LearnSubTask[] getLearnTasks() {
        return learnTasks;
    }

    public void setLearnTasks(LearnSubTask[] learnTasks) {
        this.learnTasks = learnTasks;
    }
}
